preconditions-assertj/build.gradle

```diff
@@ -0,0 +1,10 @@
+apply from: "${rootDir}/gradle/publish.gradle"
+
+dependencies {
+    api project(':preconditions')
+    api project(':safe-logging')
+
+    implementation 'org.assertj:assertj-core:3.9.1'
```

Comment:
I think this needs to be api because in order to for a consumer to compile against the `preconditions-assertj` library, they need to have assertj available at compile time (ThrowingCallable appears in a public signature).